shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -57,6 +57,12 @@
         info.surface = surface;
         info.offset = CGPointMake(layer->offset.x, layer->offset.y);
         info.zIndex = i;
+        FlutterBackingStorePresentInfo* present_info = layer->backing_store_present_info;
+        if (present_info != nullptr && present_info->paint_region != nullptr) {
+          auto paint_region = present_info->paint_region;
+          info.paintRegion = std::vector<FlutterRect>(
+              paint_region->rects, paint_region->rects + paint_region->rects_count);
```

Comment:
Not sure if it's worth commenting here (or somewhere) that this is safe because FlutterRect's size is never expected to change in the future. (It has no struct_size member as well, for that reason).